docs/_docs/08-ug-types-of-data-sources.md

```diff
@@ -0,0 +1,62 @@
+---
+title: "Types of data sources"
+permalink: /docs/ug-types-of-data-sources/
+excerpt: "Supported types of data sources"
+last_modified_at: 2021-01-28
+toc: false
+classes: wide
+---
+
+This guide helps you to utilize indexes on other types of data source layers.
+
+## Pluggable source builders
+Since Hyperspace v0.4, Hyperspace supports the indexes built on other types of data sources.
+To avoid unnecessary library dependencies, we provide pluggable source builders
+to support other types of sources.
+
+There are no special APIs for other source types, 
+so you can use the existing APIs with a DataFrame built on a different type of source seamlessly.
+
+Currently, Hyperspace supports the below source types:
```

Comment:
Do you think we should add Iceberg here and link to the on-going PR? Something like:

Iceberg (work-in-progress PR)